tools/pythonpkg/src/pyrelation.cpp

```diff
@@ -659,8 +659,14 @@ unique_ptr<DuckDBPyRelation> DuckDBPyRelation::Query(const string &view_name, co
 		auto query = PragmaShow(*rel->context.GetContext(), parameters);
 		return Query(view_name, query);
 	}
-	throw InvalidInputException("'DuckDBPyRelation.query' does not accept statements of type %s",
-	                            StatementTypeToString(statement.type));
+	{
+		py::gil_scoped_release release;
+		auto query_result = rel->context.GetContext()->Query(move(parser.statements[0]), false);
+		// Execute it anyways, for creation/altering statements
+		// We only care that it succeeds, we can't store the result
+		D_ASSERT(query_result);
```

Comment:
Should we not check the result here and throw an exception if it does not? Could we add a test with a failing query (e.g. with a table that does not exist)?

Reply:
Ah fair enough, had not thought about it, other than the assert
I blame monday
Will fix it now :)